clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -459,6 +459,9 @@ public void logUnused() {
     private  Map<String, ?> resolveConfigVariables(Map<String, ?> configProviderProps, Map<String, Object> originals) {
         Map<String, String> providerConfigString;
         Map<String, ?> configProperties;
+        Map<String, Object> originalsMutable = new HashMap<>();
```

Comment:
It's a bit weird to convert an immutable map to mutable and never convert back. Could we refactor the logic to push up the modification logic?

Reply:
@tadsul We could perhaps convert this to a immutable map and store `in `originals` like we do for `values`.

Reply:
I'd suggest naming this variable `resolvedOriginals`.